test_runner/regress/test_sharding.py

```diff
@@ -255,3 +255,26 @@ def shards_on_disk(shard_ids):
         env.neon_cli.tenant_migrate(migrate_shard, destination, timeout_secs=10)
 
     workload.validate()
+
+    # Check that we didn't do any spurious reconciliations.
+    # Total number of reconciles should have been one per original shard, plus
+    # one for each shard that was migrated.
+    reconcile_ok = env.attachment_service.get_metric_value(
+        "storage_controller_reconcile_complete_total", filter={"status": "ok"}
+    )
+    assert reconcile_ok == shard_count + split_shard_count // 2
```

Comment:
Shouldn't this be wrapped in a `wait_until`?

Reply:
No, the migrate operations above include waiting for the reconciliation to complete.